### PR TITLE
Support Code Engine service binding

### PR DESCRIPTION
**Pull request by youming-lin**

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [10.x, 12.x]
+        node-version: [14.x]
```
Comment (nfstein) [inline .github/workflows/nodejs.yml:12]: I think we're supposed to support 12 still

Comment (youming-lin) [reply .github/workflows/nodejs.yml:12]: @nfstein I've gotten Brent to update YAAS to use Node 14, and personal YAAS with this branch works.

The reason for the upgrade is that my changes use optional chaining, which is only available on 14+.

Comment (nfstein) [reply .github/workflows/nodejs.yml:12]: That makes sense, I was thinking about the skits

File: test/test-service.js [hunk 2]
```diff
@@ -106,7 +106,7 @@ function validateCreds(lang, service) {
 			, "naturalLanguageUnderstanding" , "personalityInsights" , "speechToText"
 			, "textToSpeech", "toneAnalyzer" , "visualRecognition"].includes(service)) {
 			// assert.fileContent(mappings_path, testUtils.PREFIX_SVC_BINDING_NAME + services);
-			assert.fileContent(mappings_path, `service_${SvcInfo[service]["customServiceKey"].replace(/-/g, '_')}`);
+			assert.fileContent(mappings_path, service.replace(/-/g, '_'));
 		}
 
 		if (lang == "SWIFT") {
```
Comment (nfstein) [inline test/test-service.js:112]: Is there a good reason not to test services for ce, I'm not that familiar with it? I don't see any, I'm guessing that's what's causing that tiny drop in code coverage by not testing [this line](https://github.com/IBM/generator-ibm-cloud-assets/pull/127/files?diff=unified&w=1#diff-f86a86aa4f3ceb09f975201170d902b5e392504e266e99cc5b227cfaeebaa2a0R147)

File: generators/service/enabler/index.js [hunk 3]
```diff
@@ -179,102 +177,94 @@ module.exports = class extends Generator {
...
-		});
+		this.context.addMappings(mappings);
+	}
 
-		let template = Handlebars.compile(mapping);
-		let localServiceKey = this.serviceKey;
-		let serviceKeySeparator = '_'
-		let localCredentialKeys = [];
-		let springMapping = null
-		if (this.context.application.language === "SPRING") {
```
Comment (nfstein) [inline generators/service/enabler/index.js:214]: Have you tested this against knative eventing? I'm not sure what framework specific formatting is going on here but I could see it breaking that skit.

File: generators/service/enabler/index.js [hunk 4]
```diff
@@ -179,102 +177,94 @@ module.exports = class extends Generator {
...
-			springMapping = ServiceUtils.getSpringServiceInfo(this.serviceKey)
-			if (springMapping) {
-				if (ServiceUtils.SPRING_BOOT_SERVICE_NAME in springMapping) {
-					localServiceKey = springMapping[ServiceUtils.SPRING_BOOT_SERVICE_NAME]
+	// Create mappings.json file for consuming service credentials
+	_generateMappingsJson(context, serviceId) {
+		const credentials = context?.application?.service_credentials;
```
Comment (nfstein) [inline generators/service/enabler/index.js:188]: Good method breakout and optional chaining

File: generators/service/index.js [hunk 5]
```diff
@@ -110,6 +110,81 @@ module.exports = class extends Generator {
 		return (cleanName || 'APP').toLowerCase();
 	}
 
+	_generateMappingsJson(context) {
```
Comment (nfstein) [inline generators/service/index.js:113]: I cant see where this is being called.

File: .nycrc [hunk 6]
```diff
@@ -2,7 +2,7 @@
     "check-coverage": true,
     "statements": 80,
     "branches": 65,
-    "functions": 90,
+    "functions": 89.92,
```
Comment (nfstein) [inline .nycrc:5]: Dont mind the drop in coverage but I think it would be avoided by a test with a service.

**Review by nfstein — APPROVED** [review]

**Pull request merged**